src/datasources/system/SystemDataSource.ts

```diff
@@ -19,6 +20,16 @@ export class SystemDataSource extends DataSourceApi<SystemQuery> {
     this.baseUrl = this.instanceSettings.url + '/nisysmgmt/v1';
   }
 
+  async metricFindQuery(query: MyVariableQuery, options?: any) { // GC: or query: MyVariableQuery
+    // Retrieve DataQueryResponse based on query.
+    const response = await getBackendSrv().post<{ data: MyVariableQuery[] }>(this.baseUrl + '/query-systems', { projection: "new(id, alias)" });
+  
+    // Convert query results to a MetricFindValue[]
+    const values = response.data.map(frame => ({ text: frame.id }));
+  
+    return values;
+  }
```

Comment:
Oh no I accidentally pushed the variable stuff, so I just spent the last few minutes reverting the changes

Reply:
FWIW, I noticed that you've been working out of the same branch for every PR. It may be easier to create separate branches for each change, copying pieces out of the `users/grchoi/systems-datasource` branch as needed.